pkg/controllers/management/authprovisioningv2/cluster.go

```diff
@@ -1,21 +1,38 @@
 package authprovisioningv2
 
 import (
+	"fmt"
 	"reflect"
 
 	v1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/pkg/rbac"
+	"github.com/sirupsen/logrus"
 	rbacv1 "k8s.io/api/rbac/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// OnCluster creates the role required for users assigned through PRTBs to be able to see the
+// OnCluster creates the roles required for users to be able to see/manage the
 // provisioning cluster resource
 func (h *handler) OnCluster(key string, cluster *v1.Cluster) (*v1.Cluster, error) {
 	if cluster == nil || cluster.DeletionTimestamp != nil {
-		return cluster, nil
+		return cluster, h.cleanClusterAdminRoleBindings(cluster)
```

Comment:
This needs to be split into two checks, one for nil then a 2nd for delete timestamp or the cleanClusterAdminRoleBindings will panic with nil access.

Reply:
my bad, changed it